src/Lti/Service/UserGenerator/UserGeneratorDatabaseStorage.php

```diff
@@ -24,28 +24,43 @@
 
 use Doctrine\ORM\OptimisticLockException;
 use Doctrine\ORM\ORMException;
+use OAT\SimpleRoster\DataTransferObject\AssignmentDto;
+use OAT\SimpleRoster\DataTransferObject\AssignmentDtoCollection;
+use OAT\SimpleRoster\DataTransferObject\UserDto;
+use OAT\SimpleRoster\DataTransferObject\UserDtoCollection;
+use OAT\SimpleRoster\Entity\User;
 use OAT\SimpleRoster\Lti\Model\AutogeneratedUser;
 use OAT\SimpleRoster\Model\AssignmentCollection;
 use OAT\SimpleRoster\Model\UserCollection;
-use OAT\SimpleRoster\Repository\AssignmentRepository;
 use OAT\SimpleRoster\Repository\Criteria\FindUserCriteria;
+use OAT\SimpleRoster\Repository\NativeAssignmentRepository;
+use OAT\SimpleRoster\Repository\NativeUserRepository;
 use OAT\SimpleRoster\Repository\UserRepository;
 use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 
 class UserGeneratorDatabaseStorage implements UserGeneratorStateStorageInterface
 {
-    private AssignmentRepository $assignmentRepository;
     private UserRepository $userRepository;
     private UserPasswordHasherInterface $hasher;
+    private UserDtoCollection $userDtoCollection;
+    private AssignmentDtoCollection $assignmentDtoCollection;
+    private NativeAssignmentRepository $nativeAssignmentRepository;
+    private NativeUserRepository $nativeUserRepository;
 
     public function __construct(
-        AssignmentRepository $assignmentRepository,
         UserRepository $userRepository,
-        UserPasswordHasherInterface $hasher
+        UserPasswordHasherInterface $hasher,
+        UserDtoCollection $userDtoCollection,
```

Comment:
We do not need to inject `UserDtoCollection` and `AssignmentDtoCollection`. It would be better to just create them for each `insert...` call.